### ISSUE TITLE
Add typescript support

### ISSUE THREAD

**Issue by reubenae** [intuit/user-data-for-fraud-prevention#12]
**Is your feature request related to a problem? Please describe.**
As an engineer, I want to be able to write my code in typescript, so I can find out possible bugs at compile time.

**Describe the solution you'd like**
Typescript being a supported language in this repo

**Describe alternatives you've considered**

**Additional context**

**Comment by shubhamarora** [comment]
@reubenae @skodamarthi Can I give it a shot?

**Comment by nsinghal12** [comment]
@reubenae Is this issue still available. If yes, I'd like to work on this.

**Comment by reubenae** [comment]
@nsinghal12 absolutely! We had some snagging issues on Adam's PR, which never got merged. You can refer to that for next steps if you get stuck. I recommend raising small, incremental PR's so we can avoid big merge conflicts. Looking forward to your contribution!

**Comment by regromrob** [comment]
Hi @skodamarthi is this issue apart of this year's hacktoberfest in addition to Grace Hopper?

**Comment by skodamarthi** [comment]
Hi @regromrob Yes...this is open for hacktoberfest as well. Contributions are welcome!

**Comment by regromrob** [comment]
Awesome @skodamarthi I would love to be assigned to this issue please.

**Comment by Vikki123** [comment]
Hi @reubenae @skodamarthi Would this be part of this hacktoberfest ?

**Comment by pmaidens** [comment]
@reubenae I think we can close this issue since #164 was merged. Do we also want to create a new issue to start migrating files from JavaScript to TypeScript? Or do you prefer to just have that happen naturally?

**Comment by reubenae** [comment]
I won't have access to my laptop for a couple of days so can't add an issue for the file migration - but absolutely happy for that to happen deliberately rather than over time. So if you fancy a crack at that, go ahead! Small PR's better than large ones though please!

**Comment by reubenae** [comment]
@pmaidens I'm back with laptop access now so feel free to make a PR and/or new issue as you wish, and I will review!

**Comment by reubenae** [comment]
This issue is being worked on, but there is lots of work to go around. If you're looking at this for some work, please keep your PR's small so that merge conflicts are avoided and wasted work limited.
Thanks to Peter who has been leading the charge on this!

**Comment by pmaidens** [comment]
One of the most important things left for us to do here is that we need to publish our types for external consumption, and tie that into the exported functions to ensure we maintain those contracts.

The other major piece left is to migrate tests to TypeScript as well.
